toolchain/sem_ir/facet_type_info.h

```diff
@@ -103,27 +103,76 @@ constexpr FacetTypeInfo::RewriteConstraint
 struct IdentifiedFacetType {
   using RequiredInterface = SpecificInterface;
 
-  IdentifiedFacetType() : interface_id(InterfaceId::None), num_to_impl(0) {}
+  IdentifiedFacetType()
+      : interface_id_(InterfaceId::None), num_interface_to_impl_(0) {}
 
+  auto required_interfaces() const
+      -> const llvm::SmallVector<RequiredInterface>& {
+    return required_interfaces_;
+  }
+
+  // Call `CanonicalizeRequiredInterfaces()` once the changes have been
+  // finished.
+  auto mutable_required_interfaces() -> llvm::SmallVector<RequiredInterface>& {
+    return required_interfaces_;
+  }
+
+  // Sorts and deduplicates `required_interfaces`. Call after building the sets
+  // of interfaces, and then don't mutate the value afterwards.
+  auto CanonicalizeRequiredInterfaces() -> void;
+
+  // Can this be used to the right of an `as` in an `impl` declaration?
+  auto is_impl_as_target() const -> bool { return interface_id_.has_value(); }
```

Comment:
Looking at how this is used, maybe "is_valid" is better than a declaration that is is an impl-as target.
```suggestion
  auto is_valid_impl_as_target() const -> bool { return interface_id_.has_value(); }
```